core/src/components/datetime/datetime.tsx

```diff
@@ -599,6 +599,14 @@ export class Datetime implements ComponentInterface {
   };
 
   private setActiveParts = (parts: DatetimeParts, removeDate = false) => {
+    /** if the datetime component is in readonly mode,
+     * allow browsing of the calendar without changing
+     * the set value
+     */
+    if (this.readonly) {
+      return;
+    }
```

Comment:
I noticed that if you tab to the Clear button and hit Enter, the value still clears. You might need to add another `return` like this elsewhere.

To be fair, this behavior exists in `main` too, and then some -- you can do pretty much anything you want with the keyboard controls, haha.

Reply:
We should be able to fix this by making related buttons disabled. In particular:

1. The month/year picker button
2. The cancel/clear/done buttons that amanda mentioned.
3. The time button that opens the time picker
4. The calendar day buttons

The purpose of this ticket is to make readonly behave correctly, so we should consider doing this in the PR. However, if this work drastically increases the effort required to implement this then we can do it separately since this behavior does exist in main.

Reply:
It is a decent chunk of work, because of all the items you pointed out, and some behaviors are the same for `disabled` and `readonly` while others differ. And several tests to write. But I'm mostly done with it now, so I think I might as well continue.

I did inquire about this in slack, since it felt unrelated to the ticket.